--- a/drivers/net/wwan/qcom_bam_dmux.c
+++ b/drivers/net/wwan/qcom_bam_dmux.c
@@ -367,6 +367,9 @@ static netdev_tx_t bam_dmux_netdev_start_xmit(struct sk_buff *skb,
 	if (!bam_dmux_skb_dma_submit_tx(skb_dma))
 		goto drop;
 
+	dev_sw_netstats_tx_add(netdev, 1, skb->len);
 	dma_async_issue_pending(dmux->tx);
 	return NETDEV_TX_OK;
 
@@ -533,6 +536,7 @@ static void bam_dmux_cmd_data(struct bam_dmux_skb_dma *skb_dma)
 		break;
 	}
 
+	dev_sw_netstats_rx_add(netdev, skb->len);
 	netif_receive_skb(skb);
 }
 
@@ -414,6 +418,7 @@ static void bam_dmux_netdev_setup(struct net_device *dev)
 	dev->type = ARPHRD_RAWIP;
 	SET_NETDEV_DEVTYPE(dev, &wwan_type);
 	dev->flags = IFF_POINTOPOINT | IFF_NOARP;
+	dev->pcpu_stat_type = NETDEV_PCPU_STAT_DSTATS;
 
 	dev->mtu = ETH_DATA_LEN;
